### ISSUE TITLE
feat(command): Gracefully stopping the worker

### ISSUE THREAD

**Issue by Guikingone** [Guikingone/SchedulerBundle#295]
**Situation**

Currently I'm using scheduler as separate service in my docker compose setup. All my scheduled tasks are send to the message queue. Consuming the messages is also done in a separate service. 

```yaml
# docker-compose.yml
version: '3.9'

services:
  phpfpm:
    extends:
      file: common-services.yml
      service: php-base
    restart: always

  messenger:
    extends:
      file: common-services.yml
      service: php-base
    depends_on:
      - rabbitmq
    restart: 'no'
    command: bin/console messenger:consume -vv async

  schedule:
    extends:
      file: common-services.yml
      service: php-base
    depends_on:
      - rabbitmq
    restart: 'no'
    command: bin/console scheduler:consume --wait --force -vv
```

Messenger has a command called `messenger:stop-workers`, which will gracefully stop workers by completing the currently handled messages and then stops further consuming. This is nice as I can send this command to the `messenger` container, poll the status of the container for status `exited`, and know it has completely stopped (and continue with deployment) without the actual need of stopping the container using `docker compose stop`.

Now for the `schedule` service, I'm running the consume command with `--wait --force`, which keeps the container alive and waits for new tasks to be handled (similar kind of behavior as the `messenger` service). Currently, as there is no native command for scheduler to stop (like at messenger), I am forced to stop the container directly.

**Question**

Thinking about the above, I got the following questions:

* What happens if I stop the container while at the same time a scheduled task would be started/running/active?
* Is it possible/useful to implement a [graceful stop command like in messenger](https://github.com/symfony/messenger/blob/8d36b2447ed50a92a21f031bc6d3a3fefbe73bf3/Command/StopWorkersCommand.php#L26)?

_Originally posted by @ToshY in https://github.com/Guikingone/SchedulerBundle/discussions/294_

**Comment by ToshY** [comment]
Hey @Guikingone 👋 

I know you said it's going to be released in `0.10`, but could you give an ETA of when that is? Can't wait to test it out 😃

**Comment by Guikingone** [comment]
Hi @ToshY 👋🏻 

Yes, it should be out around monday, I'm finishing some local tests 🙂
